docs/contributing/coding-guidelines/ros-nodes/topic-message-handling/index.md

```diff
@@ -0,0 +1,187 @@
+# Topic message handling guideline
+
+## Introduction
+
+Here is coding guideline for a topic message handling method which includes more enhanced method than usually used one, which is roughly explained in [Discussions page](https://github.com/orgs/autowarefoundation/discussions/4612). Refer to the page to understand the basic concept of the enhanced method.
+You can find sample source codes in [ros2_subscription_examples](https://github.com/takam5f2/ros2_subscription_examples) referred from this document.
+
+## Typical message handling method usually used
+
+At first, let us see a typical message handling method usually used.
+In a typical ROS 2 application, a topic message which is received by Subscription is referred to and processed by a dedicated callback function.
+
+```c++
+  steer_sub_ = create_subscription<SteeringReport>(
+    "input/steering", 1,
+    [this](SteeringReport::SharedPtr msg) { current_steer_ = msg->steering_tire_angle; });
+```
+
+In code above, when a topic message whose name is `input/steering` is received, an anonymous function whose description is `{current_steer_ = msg->steering_tier_angle;}` is executed as a callback in a thread. The callback function is always executed when the message is received, which leads to waste computing resource if the message is not always necessary. Besides, it costs thread-wakeup overhead just to take a message.
+
+## Enhanced method
+
+There is an enhanced method to take a message using `Subscription->take()` method only when it is needed.
+Here is a sample code to use `Subscription->take()` method executed as a timer or subscription callback to perform a main logic.
```

Comment:
I guess `it` in `before it is consumed by a main logic` means `a message`.
How about using `a message` or something instead of `it` for easy reading?